### ISSUE TITLE
Add DoorLocation to Sandbox.ModAPI.Ingame.IMyDoor

### ISSUE THREAD

**Issue by midspace** [KeenSoftwareHouse/SpaceEngineers#3]
Add the readonly property OpenPosition
to Sandbox.ModAPI.Ingame.IMyDoor

in reality it's m_currOpening.
Primarily for use in the MyAirtightDoor.

Edit:
Renamed from DoorLocation to OpenPosition for consistency with current API.

**Comment by midspace** [comment]
Just throwing this out there as a possibility.

Sandbox.Game.Entities.Blocks.MyAirtightDoorGeneric
  public virtual float DoorLocation { get { return m_currOpening; } }

Sandbox.Game.Entities.MyDoor
  public float DoorLocation { get { return m_currOpening; } }

Sandbox.Game.Entities.Blocks.MyAirtightHangarDoor
  public override float DoorLocation { get{ return m_currOpening * m_subparts.Count * m_subpartMovementDistance; } }
// inferred from UpdateDoorPosition()

**Comment by EzraBrooks** [comment]
@lord-devious I would suggest using 0 to 1 as the standard range for the property. It makes it easier for when, if ever, you're dealing with objects of multiple sizes instead of using a meter distance.

**Comment by MMaster** [comment]
@EzraBrooks I would suggest exactly the opposite because imho it's more convenient if you represent the distance in meters as you can say that one elevator floor is 0.5 / 2.5 meters every time. When you have different sizes of elevators then with approach you suggested it would mean for each elevator size (max distance) the floor positions would be different and you would need to calculate it based on max distance. Of course it's not a big deal, both approaches would work and it's just a matter of representation - one works better in some cases and the other one works better in other cases.

**Comment by lord-devious** [comment]
@EzraBrooks @MMaster Since @midspace was the original suggester, I think his opinion weighs most in this occation. Which in this case would mean meters.

**Comment by midspace** [comment]
Or as the little Taco girl says, why not both?!

**Comment by EzraBrooks** [comment]
@MMaster well, the way I see it is if you have exposed properties for the position of the door, ranging from 0 to 1, and the width of the interior of the door frame, then to get the current opening distance it would be a simple matter of multiplying the two together.  
But, as @midspace says, it's probably the most intuitive to have properties for both the opening ratio and the opening distance.

**Comment by MMaster** [comment]
I don't think there are any APIs to get max opening distance. That means that script can't figure out how far the door can go from distance and it can't figure out where the door is from ratio unless it has hard coded max. So either expose both and script will figure out max distance from ratio and distance or expose also maximum distance and then there is no reason to expose both ratio & distance. There is no reason for redundancy in programming APIs.

**Comment by lord-devious** [comment]
It's easy enough to add that property. That max should be available regardless, then we'll only need the meters.

---- Matt aka MMaster wrote ----

**Comment by MMaster** [comment]
Yes, max should be available for the elevator script to figure out how many floors are there even when elevator is at 0 distance. And since maximum is in meters I think it makes sense to have distance in meters. Counting ratio from that is easy.

**Comment by midspace** [comment]
"float OpenRatio" has been added to Sandbox.ModAPI.Ingame.IMyDoor
from change: https://github.com/KeenSoftwareHouse/SpaceEngineers/commit/2073ef1743fca874b9ac748c0ee07677e151b208#diff-4cf3a74e0c6a81a41115be05f49e9e1b

So, I still need the OpenPosition.

**Comment by MMaster** [comment]
So only Max is needed now as Position is Ratio * Max.

**Comment by midspace** [comment]
Are those subscribed to this happy with the current code I've put in pull https://github.com/KeenSoftwareHouse/SpaceEngineers/issues/172?
